Objects/abstract.c

```diff
@@ -1714,6 +1715,46 @@ PyNumber_ToBase(PyObject *n, int base)
     return res;
 }
 
+typedef struct {
+    const int slot;
+    const char name[3];
+    const int islot;
+    const char iname[4];
+} nb_info;
```

Comment:
Slots will be hot. Names will be cold, as they are only used if there is an exception. Move the names into a separate array.  If you make the slots `uint16_t`, the whole table will fit into a single cache line.

Could you separate the slots and the names into two different arrays. Unless there is an exception, the names will never be read. slots